workflow/controller/steps.go

```diff
@@ -222,6 +223,22 @@ func (woc *wfOperationCtx) executeStepGroup(ctx context.Context, stepGroup []wfv
 	// The template scope of this step group.
 	stepTemplateScope := stepsCtx.tmplCtx.GetTemplateScope()
 
+	// Sort steps by priority
+	sort.SliceStable(stepGroup, func(i, j int) bool {
```

Comment:
Perhaps we can wrap this as a function to `util/sorting`

Reply:
Put into a shared func and re-use.